packages/gensx-openai/src/index.tsx

```diff
@@ -1,68 +1,245 @@
+
+    const response = await context.client.chat.completions.create({
+      ...rest,
+      stream: true,
+      tools: openAITools,
+    } as ChatCompletionCreateParamsStreaming);
+
+    return response as Stream<ChatCompletionChunk>;
+  }
+
+  // Handle structured output case
+  if (props.structuredOutput) {
+    const { tools, structuredOutput, ...rest } = props;
+    const openAITools = tools?.map((tool) => tool.toOpenAITool());
+
+    const completion = await context.client.chat.completions.create({
+      ...rest,
+      response_format: structuredOutput.toResponseFormat(),
+      tools: openAITools,
+    } as ChatCompletionCreateParamsNonStreaming);
+
+    const content = completion.choices[0].message.content;
+    if (!content) {
+      throw new Error("No content returned from OpenAI");
+    }
+
+    try {
+      const parsed = JSON.parse(content) as unknown;
+      const validated = structuredOutput.safeParse(parsed);
+      if (!validated.success) {
+        throw new Error(
+          `Invalid structured output: ${validated.error.message}`,
+        );
+      }
+      return validated.data;
+    } catch (e) {
       throw new Error(
-        "OpenAI client not found in context. Please wrap your component with OpenAIProvider.",
+        `Failed to parse structured output: ${e instanceof Error ? e.message : String(e)}`,
       );
     }
+  }
 
-    if (props.stream) {
-      const stream = await context.client.chat.completions.create(props);
-
-      async function* generateTokens(): AsyncGenerator<
-        string,
-        void,
-        undefined
-      > {
-        for await (const chunk of stream as Stream<ChatCompletionChunk>) {
-          const content = chunk.choices[0]?.delta?.content;
-          if (content) {
-            yield content;
-          }
-        }
-      }
+  // Handle standard case (with or without tools)
+  const { tools, ...rest } = props;
+  const openAITools = tools?.map((tool) => tool.toOpenAITool());
 
-      const streamable: Streamable = generateTokens();
-      return streamable;
-    } else {
-      const response = await context.client.chat.completions.create(props);
-      const content = response.choices[0]?.message?.content ?? "";
+  return context.client.chat.completions.create({
+    ...rest,
+    stream: false,
+    tools: openAITools,
+  } as ChatCompletionCreateParamsNonStreaming) as Promise<ChatCompletionOutput>;
+});
 
-      function* generateTokens() {
-        yield content;
-      }
+// Direct OpenAI wrapper types
+type OpenAIChatCompletionProps =
+  | (Omit<ChatCompletionCreateParamsStreaming, "stream"> & { stream: true })
+  | (Omit<ChatCompletionCreateParamsNonStreaming, "stream"> & {
+      stream?: false;
+    });
 
-      return generateTokens();
-    }
+type OpenAIChatCompletionReturn<P> = P extends { stream: true }
+  ? Stream<ChatCompletionChunk>
+  : ChatCompletionOutput;
+
+// Direct wrapper over OpenAI's chat completion
+export const OpenAIChatCompletion = gsx.Component<
```

Comment:
I like the new naming of `OpenAIChatCompletion` and `ChatCompletion`. I was just thinking that each package exporting it's own thing named `ChatCompletion` was going to get very confusing.

Reply:
> I like the new naming of OpenAIChatCompletion and ChatCompletion. I was just thinking that each package exporting it's own thing named ChatCompletion was going to get very confusing.
Agree! I like this naming as well

Reply:
Hmm in theory they shouldn't have the same output type right?